tests/Aspire.Hosting.Tests/PublishAsDockerfileTests.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+using Aspire.Hosting.Utils;
+
+namespace Aspire.Hosting.Tests;
+
+public class PublishAsDockerfileTests
+{
+    [Fact]
+    public async Task PublishAsDockerFileConfiguresManifestWithoutBuildArgs()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+
+        var frontend = builder.AddNpmApp("frontend", "NodeFrontend", "watch")
+            .PublishAsDockerFile();
+
+        Assert.True(frontend.Resource.TryGetLastAnnotation<ManifestPublishingCallbackAnnotation>(out _));
+
+        var manifest = await ManifestUtils.GetManifest(frontend.Resource);
+
+        var expected =
+            $$"""
+            {
+              "type": "dockerfile.v0",
+              "path": "NodeFrontend/Dockerfile",
+              "context": "NodeFrontend",
+              "env": {
+                "NODE_ENV": "{{builder.Environment.EnvironmentName.ToLowerInvariant()}}"
+              }
+            }
+            """;
+
+        var actual = manifest.ToString();
+
+        Assert.Equal(expected, actual, ignoreLineEndingDifferences: true, ignoreWhiteSpaceDifferences: true);
+    }
+
+    [Fact]
+    public async Task PublishAsDockerFileConfiguresManifestWithBuildArgs()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+
+        var frontend = builder.AddNpmApp("frontend", "NodeFrontend", "watch")
+            .PublishAsDockerFile(buildArgs: [
+                new DockerBuildArg("SOME_STRING", "Test"),
+                new DockerBuildArg("SOME_BOOL", true),
+                new DockerBuildArg("SOME_NUMBER", 7),
+                new DockerBuildArg("SOME_NONVALUE"),
+            ]);
+
+        Assert.True(frontend.Resource.TryGetLastAnnotation<ManifestPublishingCallbackAnnotation>(out _));
+
+        var manifest = await ManifestUtils.GetManifest(frontend.Resource);
+
+        var expected =
+            $$"""
+            {
+              "type": "dockerfile.v0",
+              "path": "NodeFrontend/Dockerfile",
+              "context": "NodeFrontend",
+              "buildArgs": {
+                "SOME_STRING": "Test",
+                "SOME_BOOL": "{{bool.TrueString}}",
```

Comment:
Why isn't this:

```
"SOME_BOOL": true,
```

?

Reply:
It's treated as a string, not an actual bool with the resulting JSON. Would you recommend changing this?

Reply:
This looks extremely odd in a JSON payload. Does this really work with azd (and any other app that consumes the manifest)?

Reply:
I can update this to write it as a `bool` instead. I haven't been able to test it with `azd` as that functionality it doesn't exist yet.